### ISSUE TITLE
Row checkbox selections are inconsistent post sorting/quick-filter

### ISSUE THREAD

**Issue by yenniSuresh** [ag-grid/ag-grid#4692]
**I'm submitting a ...**  (check one with "x")
```
[] bug report => see 'Providing a Reproducible Scenario'
```

**Providing a Reproducible Scenario**
Server side rendering: When doing row checkbox selection and parallelly sorting on data, the selected rows order can change as per sorting but the selection doesn't move with actually selected row as we sort, the selection are position based, i,e row in the position of actually selected row pre sorting is shows as selected

Below is explanation in steps with a reproducible link.

Link: https://www.ag-grid.com/javascript-data-grid/server-side-model-selection/#example-checkbox

In above expand a row and select its internal rows and do sorting to see the bug.

In both behaviors imagine you have a SSR ag grid with row selection enabled along with column sorting.

**Current behavior**
1. Select the first row
2. Sort the column 
3. The first row still stay selected as per position though its not the actually selected row.

**Expected behavior**
1. Select the first row
2. Sort the column 
3. The selected row is the actual selected row should not depend on the position of selections before sort.

Above bug also happens for quick filter, if we select a row and its filtered in next step, still the visible first row post filtering stay selected.

**Comment by Jlll1** [comment]
This issue was mentioned before and it was labeled the expected behaviour for partial store. I'd love to know if it's possible to disable it from happening and handle the selection through custom implementation?

**Comment by yenniSuresh** [comment]
Hi @Jlll1 , it was possible for me to address this behaviour with a custom implementation, still I think the existing behaviour is confusing, i.e, if its not possible to show right selections upon sorting/quick-filter, it even should not show wrong rows as selections. 
A new expected behaviour for this can be to show/return possible right selection using  `getRowNodeId` for the current visible block and deselect remaining selected rows in background. This way it will go along with Partial store and also the output won't be confusing or buggy. One will have fair understanding that any selected thing which is not visible post sorting/quick-filter is deselected in background.

**Comment by Jlll1** [comment]
@yenniSuresh Correct. But in order to do custom implementation, you still need to fight this "feature", because it still will try to do its thing. What I was wondering, if there maybe is a flag (which I couldn't find myself), that'd disable this behaviour in its entirity, allowing the custom implementation to handle selection by itself.
